This patch is based on the upstream commit described below, adapted for use in
the Debian package by Peter Michael Green.

commit ac878cb6af53ff74693c25453c05570ed08d3141
Author: Jelmer VernooÄ³ <jelmer@jelmer.uk>
Date:   Sun Oct 20 18:30:29 2024 +0100

    Split out deb822-fast

Index: dep3/src/lossy.rs
===================================================================
--- dep3.orig/src/lossy.rs
+++ dep3/src/lossy.rs
@@ -18,8 +18,7 @@
 //! assert_eq!(patch_header.bug_debian, Some("https://bugs.debian.org/123456".parse().unwrap()));
 //! ```
 use crate::fields::*;
-use deb822_lossless::Paragraph;
-use deb822_lossless::{FromDeb822, FromDeb822Paragraph, ToDeb822, ToDeb822Paragraph};
+use deb822_fast::{Paragraph, FromDeb822, FromDeb822Paragraph, ToDeb822, ToDeb822Paragraph};
 
 fn deserialize_date(s: &str) -> Result<chrono::NaiveDate, String> {
     chrono::NaiveDate::parse_from_str(s, "%Y-%m-%d").map_err(|e| e.to_string())
@@ -79,7 +78,7 @@ pub struct PatchHeader {
 
 impl std::fmt::Display for PatchHeader {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        let paragraph: deb822_lossless::lossy::Paragraph = self.to_paragraph();
+        let paragraph: deb822_fast::Paragraph = self.to_paragraph();
         paragraph.fmt(f)
     }
 }
@@ -149,7 +148,7 @@ Bug-Debian: http://bugs.debian.org/51021
         );
         assert_eq!(
             header.description,
-            Some("Fix regex problems with some multi-bytes characters\n* posix/bug-regex17.c: Add testcases.\n* posix/regcomp.c (re_compile_fastmap_iter): Rewrite COMPLEX_BRACKET\nhandling.".to_string())
+            Some("Fix regex problems with some multi-bytes characters\n\n* posix/bug-regex17.c: Add testcases.\n* posix/regcomp.c (re_compile_fastmap_iter): Rewrite COMPLEX_BRACKET\nhandling.\n".to_string())
         );
     }
 
Index: dep3/Cargo.toml
===================================================================
--- dep3.orig/Cargo.toml
+++ dep3/Cargo.toml
@@ -44,7 +44,10 @@ path = "examples/read-dep3.rs"
 version = ">=0.4"
 
 [dependencies.deb822-lossless]
-version = ">=0.2"
+version = "0.3"
+
+[dependencies.deb822-fast]
+version = "0.2"
 features = ["derive"]
 
 [dependencies.url]
